src/main/scala/org/scalajs/cli/Scalajsld.scala

```diff
@@ -169,8 +170,19 @@ object Scalajsld {
 
       val linker = StandardImpl.linker(config)
       val logger = new ScalaConsoleLogger(options.logLevel)
-      val outFile = PathOutputFile(options.output.toPath())
-      val output = LinkerOutput(outFile)
+
+      val output = {
+        val js = options.output.toPath()
+        val sm = js.resolveSibling(js.getFileName().toString() + ".map")
+
+        def relURI(f: Path) =
+          new URI(null, null, f.getFileName().toString, null)
+
+        LinkerOutput(PathOutputFile(js))
+          .withSourceMap(PathOutputFile(sm))
+          .withSourceMapURI(relURI(sm))
+          .withJSFileURI(relURI(js))
```

Comment:
I believe we should only do this if `options.sourceMap` is true.

Reply:
We have explicitly allowed this (see the doc of LinkerOutput). In fact the sbt plugin itself uses this.